packages/core/src/dialects/abstract/data-types.ts

```diff
@@ -2017,14 +2019,16 @@ export class UUIDV4 extends AbstractDataType<string> {
   }
 }
 
+export type IncludeAsCallback = (includeAs: string) => [Literal | Fn, string];
```

Comment:
Technically there is nothing stopping us from accepting any sql expression, so the Expression type is a better match

```suggestion
export type IncludeAsCallback = (includeAs: string) => [Expression, string];
```

Reply:
makes sense!